cmd/metal-api/internal/datastore/migrations/08_childprefixlength.go

```diff
@@ -0,0 +1,90 @@
+package migrations
+
+import (
+	"net/netip"
+
+	r "gopkg.in/rethinkdb/rethinkdb-go.v6"
+
+	"github.com/metal-stack/metal-api/cmd/metal-api/internal/datastore"
+	"github.com/metal-stack/metal-api/cmd/metal-api/internal/metal"
+)
+
+func init() {
+	type tmpPartition struct {
+		PrivateNetworkPrefixLength uint8 `rethinkdb:"privatenetworkprefixlength"`
+	}
+	datastore.MustRegisterMigration(datastore.Migration{
+		Name:    "migrate partition.childprefixlength to tenant super network",
+		Version: 8,
+		Up: func(db *r.Term, session r.QueryExecutor, rs *datastore.RethinkStore) error {
+			nws, err := rs.ListNetworks()
+			if err != nil {
+				return err
+			}
+
+			for _, old := range nws {
+				cursor, err := db.Table("partition").Get(old.PartitionID).Run(session)
+				if err != nil {
+					return err
+				}
+				if cursor.IsNil() {
+					_ = cursor.Close()
+					continue
+				}
+				var partition tmpPartition
+				err = cursor.One(&partition)
+				if err != nil {
+					_ = cursor.Close()
+					return err
+				}
+				err = cursor.Close()
+				if err != nil {
+					return err
+				}
+				new := old
+
+				var (
+					af                       metal.AddressFamily
+					defaultChildPrefixLength = metal.ChildPrefixLength{}
+				)
+				// FIXME check all prefixes
+				parsed, err := netip.ParsePrefix(new.Prefixes[0].String())
+				if err != nil {
+					return err
+				}
+				if parsed.Addr().Is4() {
+					af = metal.IPv4AddressFamily
+					defaultChildPrefixLength[af] = 22
+				}
+				if parsed.Addr().Is6() {
+					af = metal.IPv6AddressFamily
+					defaultChildPrefixLength[af] = 64
+				}
+
+				if new.AddressFamilies == nil {
+					new.AddressFamilies = metal.AddressFamilies{}
+				}
+				new.AddressFamilies = metal.AddressFamilies{af}
```

Comment:
This is only required for private super networks and for those it's probably enough just to take the value from the partition for IPv4 and 64 for IPv6.

Reply:
Addressfamilies is set for all networks, defaultChildprefixlength only for PrivateSuper Networks